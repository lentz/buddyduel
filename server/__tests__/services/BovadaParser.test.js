const fs = require('fs');
const path = require('path');
const BovadaParser = require('../../services/BovadaParser');

describe('BovadaParser', () => {
  describe('#call()', () => {
    const bovadaJSON = JSON.parse(fs.readFileSync(path.join(
      __dirname, '..', 'sample-data', 'bovada-nfl.json',
    )));

    test('parses the JSON and returns game objects', () => {
      expect(BovadaParser.call(bovadaJSON)).toMatchSnapshot();
    });
  });
});
