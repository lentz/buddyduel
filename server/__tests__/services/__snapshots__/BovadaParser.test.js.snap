// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BovadaParser #call() parses the JSON and returns game objects 1`] = `
Array [
  Object {
    "awaySpread": 4,
    "awayTeam": "Atlanta Falcons",
    "homeSpread": -4,
    "homeTeam": "Philadelphia Eagles",
    "id": "fc4437660b1af9bff627f2559b922ab6",
    "startTime": 1536279600000,
  },
  Object {
    "awaySpread": 7,
    "awayTeam": "Houston Texans",
    "homeSpread": -7,
    "homeTeam": "New England Patriots",
    "id": "1b1b66c41e983f8523008d71f77a54e5",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": 5.5,
    "awayTeam": "San Francisco 49ers",
    "homeSpread": -5.5,
    "homeTeam": "Minnesota Vikings",
    "id": "2294b7b360150b4dfc6e0e2ff9a953d9",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": -3.5,
    "awayTeam": "Jacksonville Jaguars",
    "homeSpread": 3.5,
    "homeTeam": "New York Giants",
    "id": "97f15edb2f13d3257de013a211e9abd3",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": 2.5,
    "awayTeam": "Cincinnati Bengals",
    "homeSpread": -2.5,
    "homeTeam": "Indianapolis Colts",
    "id": "a124004a224a05a733eac1344869c5c0",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": -6,
    "awayTeam": "Pittsburgh Steelers",
    "homeSpread": 6,
    "homeTeam": "Cleveland Browns",
    "id": "bb97d6d5693347d929cbad7648e1a0b7",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": -2,
    "awayTeam": "Tennessee Titans",
    "homeSpread": 2,
    "homeTeam": "Miami Dolphins",
    "id": "76bf0888c4386d8c86f9947c9a82a56a",
    "startTime": 1536512700000,
  },
  Object {
    "awaySpread": 3,
    "awayTeam": "Kansas City Chiefs",
    "homeSpread": -3,
    "homeTeam": "Los Angeles Chargers",
    "id": "aeec512c3e74eada242a159ff85620f8",
    "startTime": 1536523500000,
  },
  Object {
    "awaySpread": 2,
    "awayTeam": "Seattle Seahawks",
    "homeSpread": -2,
    "homeTeam": "Denver Broncos",
    "id": "96ec67389f2095c8de395956a249d82c",
    "startTime": 1536524700000,
  },
  Object {
    "awaySpread": 2.5,
    "awayTeam": "Dallas Cowboys",
    "homeSpread": -2.5,
    "homeTeam": "Carolina Panthers",
    "id": "0511eec57c12e9db563f4878f3c733bd",
    "startTime": 1536524700000,
  },
  Object {
    "awaySpread": 0,
    "awayTeam": "Washington Redskins",
    "homeSpread": 0,
    "homeTeam": "Arizona Cardinals",
    "id": "3476bb9a8a145de4ab1b8a52e5a0e69a",
    "startTime": 1536524700000,
  },
  Object {
    "awaySpread": 8,
    "awayTeam": "Chicago Bears",
    "homeSpread": -8,
    "homeTeam": "Green Bay Packers",
    "id": "354f7444eb5be1bf4a768269b2a18113",
    "startTime": 1536538800000,
  },
  Object {
    "awaySpread": 7,
    "awayTeam": "New York Jets",
    "homeSpread": -7,
    "homeTeam": "Detroit Lions",
    "id": "19631f51b12579372c8ed5a5831114c3",
    "startTime": 1536621000000,
  },
  Object {
    "awaySpread": -3,
    "awayTeam": "Los Angeles Rams",
    "homeSpread": 3,
    "homeTeam": "Oakland Raiders",
    "id": "57b7ca9ba45c192a0168350a9596c622",
    "startTime": 1536632400000,
  },
]
`;
